src/sql/src/pure.rs

```diff
@@ -88,6 +99,33 @@ fn subsource_gen<'a, T>(
     Ok(validated_requested_subsources)
 }
 
+/// Generates a subsource name by prepending source schema name if present
+///
+/// For eg. if source is `a.b`, then `a` will be prepended to the subsource name
+/// so that it's generated in the same schema as source
+fn subsource_name_gen(
+    source_name: &UnresolvedObjectName,
+    subsource_name: &String,
+) -> Result<UnresolvedObjectName, PlanError> {
+    let PartialObjectName {
+        database,
+        schema,
+        item: _,
+    } = normalize::unresolved_object_name(source_name.clone())?;
+    let mut suggested_name = Vec::new();
+
+    if let Some(db_name) = database {
+        suggested_name.push(Ident::new(db_name));
+    }
+
+    if let Some(schema_name) = schema {
+        suggested_name.push(Ident::new(schema_name));
+    }
+
+    suggested_name.push(Ident::new(subsource_name));
+    Ok(UnresolvedObjectName(suggested_name))
```

Comment:
Is there a better way to get `UnresolvedObjectName` from a `PartialObjectName`?

Reply:
Not that I'm aware of. We have a `impl From<FullObjectName> for UnresolvedObjectName` here:
https://github.com/MaterializeInc/materialize/blob/1968da96d28533eb158058999c91c184e0eb64c5/src/sql/src/names.rs#L66-L76

It probably makes sense to add a `impl From<PartialObjectName> for UnresolvedObjectName` below with this logic and use that.

Reply:
Makes sense, let me add that.

Reply:
Done!